gpytorch/kernels/linear_kernel.py

```diff
@@ -47,25 +47,32 @@ class LinearKernel(Kernel):
             `len(active_dims)` should equal `num_dimensions`.
     """
 
-    def __init__(
-        self,
-        num_dimensions,
-        variance_prior=None,
-        offset_prior=None,
-        active_dims=None,
-    ):
+    def __init__(self, num_dimensions, variance_prior=None, offset_prior=None, active_dims=None):
         super(LinearKernel, self).__init__(active_dims=active_dims)
         self.register_parameter(name="variance", parameter=torch.nn.Parameter(torch.zeros(1)), prior=variance_prior)
         self.register_parameter(
             name="offset", parameter=torch.nn.Parameter(torch.zeros(1, 1, num_dimensions)), prior=offset_prior
         )
 
-    def forward(self, x1, x2):
+    def forward(self, x1, x2, batch_dims=None, **params):
+        x1_ = x1 - self.offset
```

Comment:
Given that we are hardcoding in checks for `batch_dims=(0, 2)` maybe it would at least make sense to raise a runtime error if it's something other than `(0)` or `(0, 2)`

Reply:
We do. It's in Kernel.__call__